packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -151,145 +113,209 @@ library LibProposing {
                     );
 
                     // Reward must be minted
-                    TaikoToken(resolver.resolve("taiko_token", false)).mint(
-                        input.proposer, reward
+                    LibTaikoToken.incrementTaikoTokenBalance(
+                        state, resolver, msg.sender, reward, true
                     );
                 }
             }
         }
 
-        if (_validateBlock(state, config, input, txList)) {
-            // returns true if we need to cache the txList info
-            state.txListInfo[input.txListHash] = TaikoData.TxListInfo({
-                validSince: uint64(block.timestamp),
-                size: uint24(txList.length)
-            });
-        }
+        // Initialize metadata to compute a metaHash, which forms a part of
+        // the block data to be stored on-chain for future integrity checks.
+        // If we choose to persist all data fields in the metadata, it will
+        // require additional storage slots.
+        meta.l1Hash = blockhash(meta.l1Height);
```

Comment:
looks like `meta.l1Height ` is initialized below, shall we move this line to somewhere after `meta.l1Height = uint64(block.number - 1);`?

Reply:
Good catch.